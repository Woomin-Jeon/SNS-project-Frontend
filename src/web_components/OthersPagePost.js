import React from 'react';
import ShowPost from '../components/posts/ShowPost';

function OthersPagePost({
  postState,
  setPostState,
  currentUserState,
  commentState,
  setCommentState,
  loginState,
  setLoginState,
  specificPost,
}) {
  const { post } = postState;

  return (
    <div className="timeline-post-mypage">
      {post.filter((v) => v.id === specificPost.id).length === 0
        ? (
          <div className="showpost">
            <div className="no-post">게시글이 없습니다.</div>
          </div>
        )
        : post.filter((v) => v.id === specificPost.id).map((p, index) => (
          <div key={index}>
            <ShowPost
              postState={postState}
              setPostState={setPostState}
              currentUserState={currentUserState}
              commentState={commentState}
              setCommentState={setCommentState}
              p={p}
              index={index}
              loginState={loginState}
              setLoginState={setLoginState}
            />
          </div>
        ))}
    </div>
  );
}

export default OthersPagePost;
