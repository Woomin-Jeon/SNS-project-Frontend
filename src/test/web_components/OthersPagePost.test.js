import OthersPagePost from "../../web_components/OthersPagePost";
import React from "react";
import {shallow} from 'enzyme';

describe('OthersPagePost', () => {
  let loginState;
  let setLoginState;
  let currentUserState;
  let postState;
  let setPostState;
  let commentState;
  let setCommentState;
  let specificPost;

  beforeEach(() => {
    postState = { post: [] };
  });

  it('renders well', () => {
    const component = shallow(
      <OthersPagePost
        postState={postState}
        setPostState={setPostState}
        currentUserState={currentUserState}
        commentState={commentState}
        setCommentState={setCommentState}
        loginState={loginState}
        setLoginState={setLoginState}
        specificPost={specificPost}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
