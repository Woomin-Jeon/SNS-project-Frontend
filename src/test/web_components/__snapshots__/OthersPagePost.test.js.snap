// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OthersPagePost renders well 1`] = `ShallowWrapper {}`;
